src/wp-admin/includes/class-wp-posts-list-table.php

```diff
@@ -261,23 +261,39 @@ protected function get_edit_link( $args, $link_text, $css_class = '' ) {
 		$class_html   = '';
 		$aria_current = '';
 
-		if ( ! empty( $css_class ) ) {
+		$edit_filter_link_vars = compact( 'url', 'link_text', 'css_class' );
+		/**
+		 * Filters the links created for filtering the posts list table.
+		 *
+		 * @since 6.9.0
+		 *
+		 * @param array $edit_filter_link_vars {
+		*       The edit filter link variables.
+		 *      @type string $url        The formatted link string.
+		 *      @type string $link_text  The link text.
+		 *      @type string $css_class  The class HTML.
+		 * }
+		 * @param array $args URL parameters for the link.
+		 */
+		$edit_filter_link_vars = apply_filters( 'edit_filter_links', $edit_filter_link_vars, $args );
+
+		if ( ! empty( $edit_filter_link_vars['css_class'] ) ) {
 			$class_html = sprintf(
 				' class="%s"',
-				esc_attr( $css_class )
+				esc_attr( $edit_filter_link_vars['css_class'] )
 			);
 
-			if ( 'current' === $css_class ) {
+			if ( 'current' === $edit_filter_link_vars['css_class'] ) {
```

Comment:
What if other CSS classes had been added? This seems like an ideal place to use `WP_HTML_Tag_Processor::parse_class_list()` proposed in https://github.com/WordPress/wordpress-develop/pull/10043 by @dmsnell. In that way, this code could instead be:


```suggestion
			if ( in_array( 'current', iterator_to_array( WP_HTML_Tag_Processor::parse_class_list( $edit_filter_link_vars['css_class'] ), true ) {
```

Reply:
with all of the attribute activity going on here we could also benefit at large from using the HTML API.

```php
$processor = new WP_HTML_Tag_Processor( '<a>LINK_TEXT</a>' );
$processor->next_token();
$processor->set_attribute( 'href', $edit_filter_link_vars['url'] );
$processor->set_attribute( 'class', $edit_filter_link_vars['css_class'] ?? null );
$processor->set_attribute( 'aria-current', 'page' );

$processor->next_token();
$processor->set_modifiable_text( $edit_filter_link_vars['link_text'] );

return $processor->get_updated_html();
```

Reply:
@dmsnell what about https://github.com/WordPress/wordpress-develop/pull/9950#discussion_r2445908238 ?

Reply:
well I don’t know. to be honest I didn’t fully understand the new filter, but I think the HTML API can play a nice role in constructing the HTML wherever it’s created, even if that’s before filtering.